docs/lib/predictions/fragments/js/sample.md

```diff
@@ -462,6 +460,502 @@ export default App;
+        // list of different options are here https://docs.aws.amazon.com/polly/latest/dg/voicelist.html
+      },
+    })
+      .then((result) => {
+        const audio = new Audio();
+        audio.src = result.speech.url;
+        audio.play();
+        setResponse("Generation completed");
+      })
+      .catch((err) => setResponse(err));
+  }
+
+  const setText = (value) => {
+    setTextToGenerateSpeech(value);
+  };
+
+  return (
+    <View style={styles.text}>
+      <View>
+        <Text style={styles.renderText}>Speech Generation</Text>
+        <TextInput
+          placeholder={textToGenerateSpeech}
+          onChangeText={setText}
+          style={styles.input}
+        ></TextInput>
+        <Button
+          onPress={generateTextToSpeech}
+          title="Text to Speech"
+          style={styles.button}
+        />
+        <Text>{response}</Text>
+      </View>
+    </View>
+  );
+}
+
+function TextIdentification() {
+  const [response, setResponse] = useState(
+    "You can add a photo by uploading directly from the app "
+  );
+
+  async function identifyFromFile() {
+    setResponse("identifying text...");
+
+    let permissionResult = await ImagePicker.requestCameraRollPermissionsAsync();
+
+    if (permissionResult.granted === false) {
+      alert("Permission to access camera roll is required!");
+      return;
+    }
+
+    let pickerResult = await ImagePicker.launchImageLibraryAsync();
+
+    function dataURLtoFile(dataurl, filename) {
+      var arr = dataurl.split(","),
+        checkMatch = arr[0].match(/:(.*?);/)[1],
+        str = atob(arr[1]),
+        num = str.length,
+        u8arr = new Uint8Array(num);
+
+      while (num--) {
+        u8arr[num] = str.charCodeAt(num);
+      }
+
+      return new File([u8arr], filename, { type: checkMatch });
+    }
+
+    const file = dataURLtoFile(pickerResult.uri);
+    console.log(file);
+
+    Predictions.identify({
+      text: {
+        source: {
+          file,
+        },
+        format: "ALL", // Available options "PLAIN", "FORM", "TABLE", "ALL"
+      },
+    })
+      .then(({ text: { fullText } }) => {
+        setResponse(fullText);
+      })
+      .catch((err) => setResponse(JSON.stringify(err, null, 2)));
+  }
+
+  return (
+    <View style={styles.text}>
+      <View>
+        <Text style={styles.renderText}>Identify Text</Text>
+        <Button
+          onPress={identifyFromFile}
+          title="Choose Image"
+          style={styles.button}
+        />
+        <Text>{response}</Text>
+      </View>
+    </View>
+  );
+}
+
+function EntityIdentification() {
+  const [response, setResponse] = useState("Click upload for test ");
+  const [image, setImage] = useState(null);
+
+  async function identifyFromFile() {
+    setResponse("searching...");
+    let image = await ImagePicker.launchImageLibraryAsync({
+      mediaTypes: ImagePicker.MediaTypeOptions.All,
+    });
+
+    function dataURLtoFile(dataurl, filename) {
```

Comment:
I think it would make sense to just define this once as a utility function. What do you think?